src/Storefront/Resources/app/storefront/src/plugin/listing/listing-pagination.plugin.js

```diff
@@ -129,4 +131,16 @@ export default class ListingPaginationPlugin extends FilterBasePlugin {
 
         return stateChanged;
     }
+
+    /**
+     * Update the canonical URL with the new page number.
+     * @param newPageNumber
+     * @private
+     */
+    _updateCanonicalUrl(newPageNumber) {
+        const canonicalMetaTag = document.querySelector('link[rel="canonical"]');
+        const canonicalUrl = new URL(canonicalMetaTag.href);
```

Comment:
We should add an extra error-handling/check here to verify that `canonicalMetaTag` is found in the DOM and also has the `href` is there. At the moment this can throw an error - which also happens in the jest test at the moment. 

https://github.com/shopware/shopware/actions/runs/15283422215/job/42990498187?pr=9936